### PR TITLE
Add set up instructions for pre commit hooks

### PR DESCRIPTION
**Pull request by Scrabouillmax**
Set up autoformatting on pre-commit 🎉

### CODE REVIEW

File: src/test/java/com/coincoinche/CoincoincheApplicationTests.java [hunk 1]
```diff
@@ -1,19 +1,18 @@
 package com.coincoinche;
 
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (Scrabouillmax) [inline src/test/java/com/coincoinche/CoincoincheApplicationTests.java]: This file change was done by the automated pre-commit formatting

File: README.md [hunk 2]
```diff
@@ -5,7 +5,17 @@ For the moment, this repository only contains boilerplate from Spring Boot, as w
...
+### Automatically format java files before commit.
+To format java files, we use [google-java-format](https://github.com/google/google-java-format).
+- Download google-java-format-1.7-all-deps.jar from https://github.com/google/google-java-format/releases
+- Move google-java-format-1.7-all-deps.jar at the root of the repository
+- Run `cp ./.git/hooks/pre-commit.sample ./.git/hooks/pre-commit`
+- Add the following lines to the top of the pre-commit file:
+
+`java -jar google-java-format-1.7-all-deps.jar -i $(git ls-files|grep \.java$)`
+
+`git add $(git ls-files|grep \.java$)`
```
Comment (nockty) [inline README.md]: Does this mean all `.java` files will be added automatically right before committing? Are we sure we want to recommend this practice?
For instance, say I'm working on two very different features in `FeatureOne.java` and `FeatureTwo.java`. Wouldn't I want to have two distinct commits for them?

Comment (nockty) [reply README.md]: This could also cause issues if we have some local file that we don't want to commit, `LocalMainJustForTests.java`.

**Review by nockty — CHANGES_REQUESTED** [review]
Could you double check the commands for the pre-commit hook? If I'm not mistaken, the hook automatically adds every `.java` file before committing. In my opinion, this shouldn't be recommended.

Maybe there is a way to create a script that only affects the files already staged for commit?

**Review by nockty — APPROVED** [review]

**Pull request merged**